sdk/core/Azure.Core.Extensions/src/Internal/ConfigurationClientFactory.cs

```diff
@@ -54,6 +62,62 @@ public static object CreateClient(Type clientType, Type optionsType, object opti
             throw new InvalidOperationException(BuildErrorMessage(clientType, optionsType));
         }
 
+        internal static TokenCredential CreateCredentials(IConfiguration configuration, IdentityClientOptions identityClientOptions = null)
```

Comment:
nit: Should this be `CreateCredential` since you're returning a single `TokenCredential`?